app/src/test/java/com/github/swent/echo/connectivity/NetworkServiceImplTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.swent.echo.connectivity
+
+import android.net.ConnectivityManager
+import android.net.NetworkRequest
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class NetworkServiceImplTest {
+    private lateinit var networkService: NetworkServiceImpl
+    private lateinit var connectivityManager: ConnectivityManager
+
+    @Before
+    fun setUp() {
+        connectivityManager = mockk()
+    }
+
+    @Test
+    fun `should not be online when no network`() {
+        every { connectivityManager.activeNetwork } returns mockk()
+        every { connectivityManager.getNetworkCapabilities(any()) } returns
```

Comment:
Knitpick: The following lines of code are duplicated in each test method: 
every { connectivityManager.activeNetwork } returns mockk()
        every { connectivityManager.getNetworkCapabilities(any()) } returns
            mockk { every { hasCapability(any()) } returns true }

        every {
            connectivityManager.registerNetworkCallback(
                any<NetworkRequest>(),
                any<ConnectivityManager.NetworkCallback>()
            )
        } returns Unit
Consider wrapping them in a function (part of it could be done in @Before function setup) in order to avoid code duplication.

Reply:
Yeah, you're right, I'll move it into the `@Before`!

Reply:
Actually, I couldn't use the `@Before` annotation, as I need to pass an argument to the `setUp` function. But it still makes the code much cleaner, thanks!